src/workflows/airqo_etl_utils/satellite_model_training.py

```
@@ -0,0 +1,326 @@
+            cache_key = (
+                round(float(row["latitude"]), 4),
+                round(float(row["longitude"]), 4),
+                timestamp.date().isoformat(),
+            )
+            row_data = row.to_dict()
+            try:
+                if cache_key not in cache:
+                    cache[cache_key] = cls._features_for_date(row["latitude"], row["longitude"], timestamp)
+                row_data.update(cache[cache_key])
+                row_data["sentinel2_error"] = None
+            except Exception as error:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)